collector/receiver/telemetryapireceiver/receiver_test.go

```diff
@@ -101,9 +101,10 @@ func TestHandler(t *testing.T) {
 			consumer := mockConsumer{}
```

Comment:
It seems like this feature warrants at least one basic test from json through mock `nextLogsConsumer`?

Reply:
Even something as simple as 
```
diff --git a/collector/receiver/telemetryapireceiver/receiver_test.go b/collector/receiver/telemetryapireceiver/receiver_test.go
index 0218fa3..211487a 100644
--- a/collector/receiver/telemetryapireceiver/receiver_test.go
+++ b/collector/receiver/telemetryapireceiver/receiver_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/stretchr/testify/require"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/receiver/receivertest"
 )
 
@@ -53,14 +54,21 @@ func TestListenOnAddress(t *testing.T) {
 }
 
 type mockConsumer struct {
-	consumed int
+	consumedSpans int
+	consumedLogs int
 }
 
 func (c *mockConsumer) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
-	c.consumed += td.SpanCount()
+	c.consumedSpans += td.SpanCount()
 	return nil
 }
 
+func (c *mockConsumer) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	c.consumedLogs += ld.LogRecordCount()
+	return nil
+}
+
+
 func (c *mockConsumer) Capabilities() consumer.Capabilities {
 	return consumer.Capabilities{MutatesData: true}
 }
@@ -70,6 +78,7 @@ func TestHandler(t *testing.T) {
 		desc          string
 		body          string
 		expectedSpans int
+		expectedLogs  int
 	}{
 		{
 			desc: "empty body",
@@ -95,6 +104,13 @@ func TestHandler(t *testing.T) {
 			]`,
 			expectedSpans: 1,
 		},
+		{
+			desc: "valid log",
+			body: `[
+				{"time":"2006-01-02T15:04:04.000Z", "type":"function", "record": "Sample log text"}
+			]`,
+			expectedLogs: 1,
+		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
@@ -105,12 +121,14 @@ func TestHandler(t *testing.T) {
 			)
 
 			r.nextTracesConsumer = &consumer
+			r.nextLogsConsumer = &consumer
 			require.NoError(t, err)
 			req := httptest.NewRequest("POST",
 				"http://localhost:4323/someevent", strings.NewReader(tc.body))
 			rec := httptest.NewRecorder()
 			r.httpHandler(rec, req)
-			require.Equal(t, tc.expectedSpans, consumer.consumed)
+			require.Equal(t, tc.expectedSpans, consumer.consumedSpans)
+			require.Equal(t, tc.expectedLogs, consumer.consumedLogs)
 		})
 	}
 }
```
would do the trick I think.